lib/evernote/edam/notestore/ttypes.py

```diff
@@ -1192,7 +1192,9 @@ def write(self, oprot):
             oprot.writeFieldEnd()
         if self.words is not None:
             oprot.writeFieldBegin('words', TType.STRING, 3)
-            oprot.writeString(self.words.encode('utf-8') if sys.version_info[0] == 2 else self.words)
+            if isinstance(self.words, bytes):
+                self.words = self.words.decode('utf-8')
+            oprot.writeString(self.words)
```

Comment:
This is autogenerated code by Thrift.
If you grep files by `encode('utf-8') if sys.version_info[0] == 2`, you can find many such piece of code.

So, the proper place for fixing this issue must be either Evernote thrift spec or Thrift Compiler.